### PR TITLE
Attempting to allow csproj being passed as a NuSpec file

### PR DESCRIPTION
**Pull request by jamescrowley**
First crack at fixing https://github.com/fsharp/FAKE/issues/643

### CODE REVIEW

**Comment by forki** [comment]
Thanks for the fast pull request. 

@ilkerde  could you please verify that it also works for you?

**Comment by ilkerde** [comment]
@forki Thanks for letting me know. @jamescrowley Thanks for your fast contribution. I like your changes since the case of accompanying .nuspec is quite often.

However, I checked this PR and have to say that it breaks a test. After running CI I receive

    Failures:
    when packing with nuspec template
    » should create nupkg file (FAIL)
    System.Exception: Error during NuGet package creation. C:\Code\ilkerde.fake\test\TestData\nuget.exe pack -Version 0.0.1 -OutputDirectory "C:\Users\ilker.cetinkaya\AppData\Local\Temp" "fake.nuspec"

This is the test for `NugetPack` to be able to transform templates. Seems that your change breaks this. I suppose that you are changing something along the line of `createNuspecFile`, most probably the return value around [L227](https://github.com/jamescrowley/FAKE/blob/fix.NuGetPack/src/app/FakeLib/NuGet/NugetHelper.fs#L227). I would recommend to have tests for your case (nuspec alongside the csproj) as well.

Any chance to get this fixed (and probably even have a test for it)?

**Comment by forki** [comment]
@jamescrowley can I help you to get this in?

**Comment by jamescrowley** [comment]
@forki This isn't good to merge btw, the whole approach doesn't work - requires a bigger refactor to deal with issues like the current code just tries to delete whatever file you pass in as an argument. And also doesn't work if you pass in a .csproj with a .nuspec alongside that's a template.

**Comment by jamescrowley** [comment]
I think this at least works now @forki @ilkerde. it's ugly as anything, but has a test and will have to defer to others in terms of how we tidy up

File: src/test/Test.FAKECore/Test.FAKECore.csproj [hunk 1]
```diff
@@ -225,6 +225,12 @@
     <Content Include="TestData\AllObjects_2.txt">
       <CopyToOutputDirectory>Always</CopyToOutputDirectory>
     </Content>
+    <Content Include="TestData\bin\Debug\TestNuGetPack.dll">
```
Comment (forki) [inline src/test/Test.FAKECore/Test.FAKECore.csproj]: I think these files are gitignored. You need to add them "forced"

**Comment by jamescrowley** [comment]
https://bugzilla.xamarin.com/show_bug.cgi?id=17430 Hmm. Nice. So you can't actually nuget pack on a csproj file on mono. Thoughts @forki ?

**Comment by forki** [comment]
mhm. then maybe just use isMono and throw a good exception.
In the test try to catch that exception in the mono case. does this sound OK?

BTW: we are actively working on http://fsprojects.github.io/Paket/paket-pack.html

**Comment by jamescrowley** [comment]
@forki I think I'd probably make the test check for isMono rather than the nuget helper? given we don't actually know whether nuget.exe in the future would support, and at least you get a decent error output when it fails ow?

**Comment by forki** [comment]
yes that's ok, too

**Comment by forki** [comment]
thanks for being so patient ;-)

**Pull request merged**